### PR TITLE
Fix map ttl test

### PR DESCRIPTION
**Pull request by mmedenjak**
The test could fail if the assertion that the entry was in the map was executed too late. Since this is an expiration test, the assertion was changed to just wait until the entry was null.

Fixes: https://github.com/hazelcast/hazelcast/issues/8811

### CODE REVIEW

File: hazelcast/src/test/java/com/hazelcast/map/impl/tx/MapTransactionTest.java [hunk 1]
```diff
@@ -408,14 +413,19 @@ public Boolean execute(TransactionalTaskContext context) throws TransactionExcep
         });
         assertTrue(b);
 
-        IMap map1 = h2.getMap("putWithTTL");
-        assertEquals("value", map1.get("1"));
+        final Object result = map2.get("1");
+        if (deletedCounter.get() > 0) {
+            System.out.println("Value was already removed from map, quitting test");
```
Comment (eminn) [inline hazelcast/src/test/java/com/hazelcast/map/impl/tx/MapTransactionTest.java]: In the test case we are expecting the value to be evicted eventually, so we don't need to test against value insertion but value eviction.

So we can refactor the remaining test to an `assertTrueEventually` block which checks the `deletedCounter.get()>0` and `assertNull(map2.get("1"))`. This way we can also get rid of the `sleep`

**Comment by mmedenjak** [comment]
run-lab-run

File: hazelcast/src/test/java/com/hazelcast/map/impl/tx/MapTransactionTest.java (hunk 1 above)
Comment (mmedenjak) [inline hazelcast/src/test/java/com/hazelcast/map/impl/tx/MapTransactionTest.java]: This was changed. Also, the atomic counter was removed due to the fact that the interceptor can be serialized and invoked on an another instance, thus failing to pass assertion. As the counter was irrelevant and we already have one assertion that the map contains the value, we just need to assert eventually that the map does not contain the value anymore. WDYT?

**Comment by mmedenjak** [comment]
run-lab-run

**Comment by mmedenjak** [comment]
@eminn can you check if it looks ok now?

**Comment by jerrinot** [comment]
run-lab-run

**Pull request merged**